src/Compilers/VisualBasic/Portable/Symbols/Source/SourceAssemblySymbol.vb

```diff
@@ -1042,8 +1042,14 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Symbols
                 Dim verString = DirectCast(attrData.CommonConstructorArguments(0).ValueInternal, String)
                 Dim version As Version = Nothing
                 If Not VersionHelper.TryParseAssemblyVersion(verString, allowWildcard:=Not _compilation.IsEmitDeterministic, version:=version) Then
-                    diagnostics.Add(ERRID.ERR_InvalidVersionFormat, GetAssemblyAttributeFirstArgumentLocation(arguments.AttributeSyntaxOpt))
+                    Dim attributeArgumentSyntaxLocation As Location = GetAssemblyAttributeFirstArgumentLocation(arguments.AttributeSyntaxOpt)
+                    If _compilation.IsEmitDeterministic AndAlso verString.Contains("*"c) Then
```

Comment:
The C# equivalent considers that `verString` can be `null`. Should replicate that check here or attempt to remove it in C#

Reply:
Thanks. I have added an identical null check to remain consistent.